roles/openshift_health_checker/test/docker_image_availability_test.py

```diff
@@ -3,26 +3,171 @@
 from openshift_checks.docker_image_availability import DockerImageAvailability
 
 
-@pytest.mark.xfail(strict=True)  # TODO: remove this once this test is fully implemented.
-@pytest.mark.parametrize('task_vars,expected_result', [
-    (
-        dict(
-            openshift=dict(common=dict(
-                service_type='origin',
-                is_containerized=False,
-            )),
-            openshift_release='v3.5',
-            deployment_type='origin',
-            openshift_image_tag='',  # FIXME: should not be required
-        ),
-        {'changed': False},
-    ),
-    # TODO: add more parameters here to test the multiple possible inputs that affect behavior.
+@pytest.mark.parametrize('deployment_type,is_active', [
+    ("origin", True),
+    ("openshift-enterprise", True),
+    ("enterprise", False),
+    ("online", False),
+    ("invalid", False),
+    ("", False),
 ])
-def test_docker_image_availability(task_vars, expected_result):
+def test_is_active(deployment_type, is_active):
+    task_vars = dict(
+        deployment_type=deployment_type,
+    )
+    assert DockerImageAvailability.is_active(task_vars=task_vars) == is_active
+
+
+@pytest.mark.parametrize("is_containerized", [
+    True,
+    False,
+])
+def test_all_images_available_locally(is_containerized):
+    def execute_module(module_name, args, task_vars):
+        assert module_name == "docker_image_facts"
+        assert 'name' in args
+        assert args['name']
+        return {
+            'images': [args['name']],
+        }
+
+    result = DockerImageAvailability(execute_module=execute_module).run(tmp=None, task_vars=dict(
+        openshift=dict(common=dict(
+            service_type='origin',
+            is_containerized=is_containerized,
+        )),
+        deployment_type='origin',
+        openshift_release='v3.4',
+        openshift_image_tag='3.4',
+    ))
+
+    assert not result.get('failed', False)
+
+
+def test_all_images_available_remotely():
+    def execute_module(module_name, args, task_vars):
+        if module_name == "docker_image_facts":
```

Comment:
The current code is fine, but I'd like to show an alternative that shows one "trick" to eliminate chains of ifs:

```python
def execute_module(module_name, args, task_vars):
    return {
        'docker_image_facts': {'images': []},
        'docker_info': {'info': {'Registries': [{'Name': 'docker.io'}, {'Name': 'registry.access.redhat.com'}]}},
    }.get(module_name, {'changed': False})
```

Because it is unlikely that we add more items to the list of Registries in the docker_info output, it is rather okay to have it in a single line, though it could it one registry per line also if we want to make that easier to read.